### PR TITLE
jsp with rest

### PR DESCRIPTION
**Pull request by un-install**

### CODE REVIEW

**Comment by un-install** [comment]
https://github.com/un-install/webapp/blob/master/src/main/resources/jspscsh/page-1.PNG

**Comment by un-install** [comment]
https://github.com/un-install/webapp/blob/master/src/main/resources/jspscsh/page-2.PNG

File: src/main/java/lesson25/MateRest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    @GET
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    @Produces(MediaType.TEXT_HTML)
+    @Path("/students")
+    public Viewable stud(@QueryParam("yearOfBorn") String yearOfBorn) {
+        if (yearOfBorn != null && mateGroup.getStudents().stream().anyMatch(s -> yearOfBorn.equals("" + s.getYearOfBorn()))) {
+            Person student = mateGroup.getStudents().stream().filter(s -> yearOfBorn.equals("" + s.getYearOfBorn())).collect(Collectors.toList()).get(0);
+            return new Viewable("/stud-resp", student);
+        }
+        return new Viewable("/stud");
```
Comment (Cherkashyn) [inline src/main/java/lesson25/MateRest.java]: Maybe here you need return new Viewable("/stud-resp"); ??

Comment (un-install) [reply src/main/java/lesson25/MateRest.java]: na-ah

**Review by Cherkashyn — APPROVED** [review]
Ok

File: src/main/java/lesson25/MateRest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        Object o = new Object();
+        List<Object> some = (ArrayList) o;
+        if (groupId != null && groupId.equals("" + mateGroup.getId())) {
+            return new Viewable("/index", mateGroup);
+        }
+        return new Viewable("/index");
+    }
+
+    @GET
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
```
Comment (SerhiiPasko) [inline src/main/java/lesson25/MateRest.java]: Is it need?

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Add screenshots. Also you must create your  model with some info - don't use my "mateGroup"

**Comment by un-install** [comment]
improvements added

**Pull request merged**